modzy/jobs.py

```diff
@@ -447,7 +453,25 @@ def submit_files_bulk(self, model, version, sources, explain=False):
             raise
         return Job(open_job, self._api_client)
 
-    def submit_aws_s3(self, model, version, source, access_key_id, secret_access_key, region, explain=False, source_name='job'):
+    def _append_input(self, job, input_item_name, data_item_name, input_value, chunk_size):
+        iterable = None
+        if isinstance(input_value, (bytes, bytearray)):
+            iterable = bytes_to_chunks(input_value, chunk_size)
+        else:
+            iterable = file_to_chunks(input_value, chunk_size)
+            self.logger.debug(f"_append_input -> {iterable}")
+        i = 0
+        for chunk in iterable:
```

Comment:
instead of using your own counter, I think you can just do a enumerate here.

for i, chunk in enumerate(iterable):
    ...

Reply:
I have a question here: if the iterable object is a generator will the enumerate execute all the yields to get the total count?